src/sentry/models/integration.py

```diff
@@ -87,3 +87,29 @@ def add_organization(self, organization_id, default_auth_id=None, config=None):
             return False
         else:
             return True
+
+    def add_project(self, project_id, config=None):
+        """
+        Add a project to this integration. Requires that a
+        OrganizationIntegration must exist before the project can be added.
+
+        Returns True if the ProjectIntegration was created
+        """
+        from sentry.models import Project
+        if not OrganizationIntegration.objects.filter(
+            organization_id=Project.objects.get(id=project_id).organization_id,
```

Comment:
you could do `organization_id=Project.objects.filter(id=project_id).values_list('organization_id', flat=True).get()` for a slightly more efficient query (just selecting org id rather than the whole object)

Reply:
OR drop the `.get()` and it'd just be a subquery. :)

Reply:
even better 👌

Reply:
71ec9d3